py/compile.c

```diff
@@ -2115,7 +2115,13 @@ STATIC void compile_namedexpr_helper(compiler_t *comp, mp_parse_node_t pn_name,
     qstr arg = MP_PARSE_NODE_LEAF_ARG(pn_name);
     compile_node(comp, pn_expr);
     EMIT(dup_top);
+    scope_t *old_scope = comp->scope_cur;
+    if (SCOPE_LIST_COMP <= comp->scope_cur->kind && comp->scope_cur->kind <= SCOPE_GEN_EXPR) {
```

Comment:
The correctness of this condition is dependent on the exact ordering of some `enum` values in another file, which makes me fear that someone will break it down the line. Maybe it's a stylistic thing but I think it would make sense to define a macro called something like `SCOPE_IS_COMP_LIKE`, next to where you define `SCOPE_IS_FUNC_LIKE` and the `enum` itself in `scope.h`, that encompasses the correct condition and then use that macro here. That way if one day someone changes the values in the `enum` they will know to update those adjacent macros.

Reply:
>  I think it would make sense to define a macro called something like `SCOPE_IS_COMP_LIKE`, next to where you define `SCOPE_IS_FUNC_LIKE`

Yes I completely agree.

Reply:
Now done.